sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/EventTimeWatermark.scala

```diff
@@ -72,3 +74,32 @@ case class EventTimeWatermark(
   override protected def withNewChildInternal(newChild: LogicalPlan): EventTimeWatermark =
     copy(child = newChild)
 }
+
+case class UpdateEventTimeWatermarkColumn(
+    eventTime: Attribute,
+    delay: CalendarInterval,
+    child: LogicalPlan) extends UnaryNode {
+  override def output: Seq[Attribute] = child.output.map { a =>
```

Comment:
Function is effectively same as the `EventTimeWatermark` case ? can we just have them share the same private function ?

Reply:
Yep, done.